pkg/services/ngalert/store/alertmanager_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build integration
+// +build integration
+
+package store
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAlertManagerHash(t *testing.T) {
+	sqlStore := sqlstore.InitTestDB(t)
+	store := &DBstore{
+		SQLStore: sqlStore,
+	}
+	t.Run("After saving the DB should return the right hash", func(t *testing.T) {
+		config, configMD5 := "my-config", fmt.Sprintf("%x", md5.Sum([]byte("my-config")))
```

Comment:
Talked about this offline, it's not supported by all databases we use (i.e. SQLite)

Reply:
Sorry, I meant at the store code layer. I.e. not have the DB itself compute the hash, but have the calls to md5.Sum() (currently in notifier/alertmanager.go) instead live in store/alertmanager.go - baking this into the store layer API. I think this is a better separation of concerns because the way we're computing the hash isn't app logic, and would better belong in an infrastructure package. It also single-sources it and makes sure you can't pass in a bad/incorrect hash.

Reply:
@alexweav good catch, now I got it :-D will change it 👍